mautic/mautic
# [UI/UX] Toolbar page actions

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | 🔴 
| New feature/enhancement? (use the a.x branch)      | 🟢
| Deprecations?                          | 🔴
| BC breaks? (use the c.x branch)        | 🔴
| Automated tests included?              | 🔴 
| Related user documentation PR URL      | mautic/user-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation-new#... 
| Issue(s) addressed                     | Fixes #... 



## Description

This PR groups related elements together to follow UX best practices.

When working on a details page, it makes sense to have a page header containing actions for the current resource (e.g. Edit action, to edit the current form, for example). 

Maybe not that much for a page that lists several resources for the same thing... In a first moment, you might look at the current state of our index pages and think that they look good.

![image](https://github.com/user-attachments/assets/8fd56cd5-d8c9-4e4d-8d32-7d239c55e60e)

but what if we add a widget to show a chart about your lead generation? what would be the "New" page action for, creating a contact or adding a new widget?

as we think about making the listing pages more attractive, is inevitable to consider adding content that creates a relationship with everything else you do in Mautic everyday

Placing the page actions on the toolbar specifically for listing pages aim to address this case, allowing us to keep developing content to make the tool get increasingly better without causing confusion to our users

![image](https://github.com/user-attachments/assets/19d875ac-4a24-4376-bafa-503b2f3dd951)







---
### 📋 Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Open Contacts and have a look at the list toolbar, also interact
3. Repeat for all other listing pages
